### DISCUSSION TITLE
A thesis using WIDesigner

### DISCUSSION THREAD

**Discussion by chihowkwan** [edwardkort/WWIDesigner#153] [Show and tell]
Hello! I would like to share with you my just published master’s thesis which uses the flute study in WIDesigner. The topic is “analyzing and improving the tuning of Ding, Xilin’s xindi using WIDesigner” and I wrote in both Chinese and English for better introducing this research to people. The text can be downloaded in the link below and there is also some basic information. It is from an official website of theses in Taiwan.

https://ndltd.ncl.edu.tw/cgi-bin/gs32/gsweb.cgi?o=dnclcdr&s=id=%22111TNCA0750003%22.&searchmode=basic

WIDesigner plays a key role in calibrating, designing, and predicting in my research. But I also explain the background understanding and methodology necessary for improving and analyzing the tuning of my research subject: xindi, a Chinese transverse flute designed nearly 90 years ago.

I would like to express my gratitude to developers of WIDesigner which also appears in my acknowledgements. I hope this program can continue to grow!

**Comment by bhp1** [comment]
You're welcome, Chihow.  Glad you found it useful.  An 11-hole chromatic instrument sounds like a demanding target to model.  Did you find that WIDesigner gave accurate tuning predictions?  In the WIDesigner options, what Blowing Level did you set on the whistle/flute tab?

**Comment by chihowkwan** [comment]
About the question of whether the predictions are accurate, though I would like to answer yes simply, I think of many details in my research. Therefore, I need to briefly describe the process. I went through two attempts of making my improved versions of the xindi. In the first attempt, I used a conjectural airstream length and a beta factor 0.4 suggested by WIDesigner to optimize the instruments. I used my own playing and instruments printed in this attempt to calibrate my blowing parameters. I described a difference observed between WID’s calibrated prediction and my playing result in one of the instruments and suggested possible reasons. However, this may be just a small detail in my research. It might be what‘s in the second attempt which is more important.

I used my calibrated blowing parameters to optimize and make instruments in the second attempt. My playing results correspond quite well with WIDesigner’s predictions this time. Then, I invited nine players (one by one and in different time and space) to test these instruments and recorded their playing results. Their original results showed no clear patterns even with simple temperature corrections, possibly indicating everyone uses different blowing parameters and my calibration is only effective to myself. I needed to use a statistical way which is to get each one’s deviation from mean to see possible relative tuning patterns. This way really showed patterns (at least to my eyes). Then I calculated the average of these deviations from means and used this average to calibrate blowing parameters and asked WID to predict the tuning with these parameters. High similarities were seen in these comparisons, and in this way WID is accurate. One minor difference which I also described in the thesis is “the D5 of the average players’ playing was slightly sharper than the program’s prediction on the flared, the tapered, and the cylindrical version. It itself is the mode 2 vibration of the lowest note, which is often played with adding pressure in my experience of observing flute players including myself. Probably because it is a note that uses the full effective vibrating length of the flute and also in the middle register which is easy to play, people tend to overdo it. I assume WIDesigner’s prediction does not include this tendency” quoted from my thesis. The xindi I made has a range similar to the flute, so you can roughly consider its D5 fingering to be the D5 of the Boehm or Baroque flutes.

In addition, all WIDesigner calibrations, predictions, and my related comparisons are done within the range D4–E6 of the xindi. In my experience, including higher pitches makes predictions unrealistic probably because people use non-linear blowing on these notes but WIDesigner can only predict in a linear way, I think.

About the blowing level, I set it to the WID’s suggested 9 at all time. It is one of the parameters which remains the same in my research. I just discovered in my thesis I accidentally call it “playing strength” one time and I missed correcting. I may be that I translated the blowing level to Chinese in my mind and then translated back to English, causing the name confusion! Playing strength might possibly be what I understand about the blowing level, though it might be quite superficial.

**Reply by bhp1** [reply]
Regarding the differences in playing results from different players, possible explanations include:

- Weaknesses in the WIDesigner model.  Side-blown flutes give the player control over a lot of variables that WIDesigner assumes are static, like distance between the lips (the airway exit) and the edge of the embouchure hole (the labium).
- Some players instinctively adjust their playing to keep individual notes in tune.  Did you notice that some players did a better job at the relative tuning between notes?
- WIDesigner's simplistic set of blowing patterns may not capture the full range of player preferences.

You can test the third explanation by entering a player's actual playing frequencies into a tuning file, and looking at [the tuning graph](https://github.com/edwardkort/WWIDesigner/wiki/Evaluating-Instruments-with-the-Tuning-Table-and-Tuning-Graph#the-tuning-graph). If the green (or yellow or red) markers make a regular curve of any sort then the player is exhibiting a preferred blowing pattern.  If WIDesigner could model that blowing pattern, it could accurately predict that player's playing frequencies.  More information about the blowing pattern preferences of real players would be valuable.  Your comment about D5 is very interesting.  You may be right that players like to give D5 a little extra push, or perhaps all the other flutes they've played have needed that extra push.  WIDesigner aims for a smooth change through C5-D5-E5, and most instruments probably don't meet that target.

"Playing strength" is a reasonable translation of the blowing level, and not much different than I have thought of it.

**Comment by chihowkwan** [comment]
About noticing some better tuning results, no, I actually did not think of noticing them. I discussed mainly results of the average tuning patterns of these players. Presented as line graphs, individual tuning patterns formed messy looks. I originally thought that it is less meaningful to discuss individual results than overall patterns. Since you asked this question, after some comparisons, I found that I myself did a better job on corresponding my tuning patterns with WIDesigner predictions on these instruments I made, and this also means my relative tuning is better.

![me playing tapered](https://user-images.githubusercontent.com/85783979/217431963-2cff1384-47d7-40e1-b552-875feedc3fa5.png)
Fig. 1. Tuning graph of my playing (with the calculation of deviation from mean) of the tapered version xindi made in the second attempt of the thesis

![no 4 playing tapered](https://user-images.githubusercontent.com/85783979/217432012-4da24d52-5ad4-469b-b793-15b1d2df3102.png)
Fig. 2. Tuning graph of player 4’s playing (with the calculation of deviation from mean) of the same instrument of fig. 1

![螢幕擷取畫面 (194)](https://user-images.githubusercontent.com/85783979/217432061-6a4613f0-57ee-4117-aeda-0bef4f8f5676.png)
Fig. 3. Deviations from means of player 4, me, WID prediction (calibrated with the deviation average of nine players) and the deviation average of nine players not including me on the same instrument

In fact, I did not think of the tuning graph function to be able to compare actual recorded tuning patterns with WIDesigner preferred ones until now. Previously, I only use it to check if a newly designed instrument has its green dots beautifully overlap with blue circles, and this means it is an instrument easily played in tune.

In fig. 1, I used my playing frequencies on one tapered version designed for improving the xindi to form a tuning graph. Since they are originally only for proving this flute made in the second attempt is basically in tune with my blowing, I did not recorded notes higher than E6. Fig. 2 and 3 includes notes up to B6. Fig. 2 is a demonstration of a player’s result I randomly choose to form the tuning graph here. Compared with fig. 1, fig. 2 is more deviated from blue circles. I must indicate that almost all players demonstrate this deviation but each has his/her own pattern. Fig. 3 uses the original method I use in the thesis to present tuning patterns. For clarity’s sake, only player 4’s and my individual results are shown here. It is a mysterious thing that the deviation average of nine players are closer to the WIDesigner prediction than individual results.

I think the tendency that my result is relatively closer to the WID prediction than individual results of these nine players may come from two factors. Because I made these instruments, I could be more familiar with them than others. With some previous experience of calibrating and testing WIDesigner-designed instruments, I was also more familiar with its blowing method. Indeed, I tried to give indications to these players which suggest WID’s blowing method (I think so): “Written indications for players are ‘please play the chromatic scale at a comfortable volume (medium volume), without looking at tuners, for 1–2 minutes. The playing time for each note is about 0.25–1 second.’ Although they do not directly suggest the default WIDesigner method of increasing the air pressure with higher pitches, they indirectly make players approach this linear blowing method by suggesting chromatic scales where pitches change gradually, normal volume for notes, playing without too much tuning control, and duration neither too long nor too short” (from my thesis). However, I really could not expect one to change one’s playing habits within an hour.

Since every player can have his/her own playing habits, the methodology of treating the average of their deviations from means as an instrument’s actual tuning pattern might be effective as long as there are sufficient players to test to minimize individual differences. Some might think nine are not enough for showing statistical significance. This is due to my limitation of time and connections. I expect better research in the future.

I have a question. Because I went to WIDesigner to generate the above fig. 1 and 2, I made quick calibrations using those frequencies and saw resulting beta factors and air stream lengths. And these graphs are drawn based on these calibrated models. I also input others’ results which I did not show here. As I have mentioned, their green dots are all obviously not in a good curve shape. About the parameters, I noticed different numbers in the airstream length with different players. However, regarding the beta factor, the number 0.20000 almost appears all time in results of individual players (myself excluded). Is 0.20000 a number demonstrating a specific genre of playing? I actually can’t understand the beta factor—the jet spatial amplification coefficient—even though I found related references. I did not have backgrounds in science and mathematics, and this concept seems too difficult for me.

**Comment by bhp1** [comment]
The beta factor is a parameter that influences the calculation of loop gain, which determines the predicted minimum playing frequency for each note.  In general terms, a larger beta means larger loop gains, which means lower predicted minimum frequencies because the loop gain will stay above the threshold of 1 with lighter blowing.

The value of 0.20000 is an artifact of the default constraints on the flute calibrator, nothing more.  When I implemented the flute calibrator, I didn't think beta would likely go below 0.2, so I made that the lower bound in the default constraints.  This now looks inappropriate; like you, I have seen calibration lead to beta values of 0.2 or lower, although I'm not sure why yet.

If you select an instrument, a tuning, and Flute Calibrator, then select from the menu File->Create default constraints, you'll get a new tab with lower and upper bounds for airstream length and beta; the lower bound for beta will be 0.2000.  You can change that value to 0.1 or even 0.0, and run an optimization with those constraints; the resulting instrument tab will (likely) have a beta factor less than 0.2.  

When you produced Fig 2 above, did you run the Flute Calibrator on the actual playing frequencies of Player 4 first?  If so, please re-display the tuning graph for Player 4 _using the same airstream length and beta that you used for yourself (Fig 1)._  The two graphs should look identical except for the positions of the green (or yellow or red) markers.

I think your methodology of averaging the tuning over a number of players is sound.  It might be interesting to calibrate an instrument using the average tuning, and then look at the tuning graphs of individual players for that calibration.

**Comment by chihowkwan** [comment]
Thank you for pointing out the Flute Calibrator has constraints to set! I only use constraints of optimizers and did not discover that it is also in this function until now.

Fig. 4 is the tuning graph drawn from your indication, and fig. 5 is a comparison I made between two graphs. The calibration of player 4 results in an airstream length less than mine and a beta factor slightly higher than mine. I noticed a general position shift of green dots in the first octave. Actually, I don’t know how to interpret fig. 4. It is player 4’s playing results with my airstream length and beta factor. I don’t know how this can be meaningful for designing or testing.

In addition, I figure out that whether a note has weight influences the route blue circles display. These figures here contain notes which have weights from D4 to E6. Notes higher than E6 have 0 weight. If these extremely high notes are accidentally set to have weights, blue circles choose a different route which goes generally towards fmax. I think this is quite reasonable.

![No 4 playing with my parameters (2)](https://user-images.githubusercontent.com/85783979/218975826-f0e69720-83a4-42dc-a75b-8044abd1d9ed.png)
Fig. 4. Tuning graph of player 4’s playing (with the calculation of deviation from mean) under my airstream length and beta factor

![螢幕擷取畫面 (199)](https://user-images.githubusercontent.com/85783979/218975920-4123dcbb-c579-4f2c-9952-3b2a8811dec9.png)
Fig. 5. Left: tuning graph of player 4 using my airstream length and beta factor. Right: tuning graph of player 4 under its resulted calibration (its airstream length and beta factor)

**Comment by bhp1** [comment]
> I figure out that whether a note has weight influences the route blue circles display. 

Yes, particularly at blowing level 9, WIDesigner picks a blowing pattern that will put the lowest note close to its fmax, and the highest _weighted_ note close to its fmin, and a smooth curve for all of the notes in between, weighted or not.  To get a useful blowing pattern (the blue circles), you need to be very careful about which note is the highest note with non-zero weight.

From the new Figure 4, Player 4's blowing pattern does have some regularity: fairly soft on the first octave, and much harder on the second and third octaves.  I'm going to hazard a wild guess that Player 4 is less experienced than you, and doesn't realize that you can push the first octave a bit harder, and don't have to push the second octave that hard.  In general, they don't manage the register transitions C5-C#5-D5-D#5-E5, or C6-C#6-D6-D#6-E6 very well.  Or perhaps it's just inexperience with this particular flute.

Because Player 4 doesn't handle this flute well, the flute calibration with their tuning doesn't have much value.  The calibration assumes the player is aiming for a regular blowing pattern.  For Player 4, that is a bad assumption.
